apps/web/src/app/tracks/_components/track-details.tsx

```diff
@@ -0,0 +1,112 @@
+'use client';
+
+import { useSession } from '@repo/auth/react';
+import { toast } from '@repo/ui';
+import { useRouter } from 'next/navigation';
+import { useEffect, useState } from 'react';
+import { EnrollButton } from './enroll-button';
+import { enrollUserInTrack, getTrackDetails, unenrollUserFromTrack } from './track.action';
+
+interface TrackDetailProps {
+  // trackid
+  slug: string;
+}
+
+export function TrackDetail({ slug }: TrackDetailProps) {
+  const session = useSession();
+  const router = useRouter();
+  const trackId = parseInt(slug);
+  const [trackChallenge, setTrackChallenge] = useState<Awaited<
+    ReturnType<typeof getTrackDetails>
+  > | null>(null);
+  const [isEnrolled, setIsEnrolled] = useState(false);
+
+  // things you need to do to skip first render & call server action on client component.
+  // this is nightmare fuel.
+  // todo: find better way
+  useEffect(() => {
```

Comment:
why cant this be fetched in the RSC and fed in?

Reply:
i simplified it a lot trash, we don't have this mess anymore. : D